File: backend/Plant_tracker/src/main/java/com/example/Plant_tracker/models/Action.java
package com.example.Plant_tracker.models;

import jakarta.persistence.*;
import java.time.LocalDateTime;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;


@Entity
@Getter
@Setter
@NoArgsConstructor
public class Action {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)  
    private Long idAction;
    private String type;
    private LocalDateTime date;

    @ManyToOne
    @JoinColumn(name = "plant_id", referencedColumnName = "id")  
    private UserPlant plant;  
}

File: frontend/app/src/main/java/com/example/planttrackerapp/backend/domain/Action.kt
//package com.example.planttrackerapp.domain;
//
//
//import androidx.room.*
//import java.time.LocalDateTime
//
//@Entity(
//    tableName = "actions",
//    foreignKeys = [
//        ForeignKey(
//            entity = UserPlant::class,
//            parentColumns = ["id"],
//            childColumns = ["plant_id"],
//            onDelete = ForeignKey.CASCADE
//        )
//    ],
//    indices = [Index(value = ["plant_id"])]
//)
//data class Action(
//    @PrimaryKey(autoGenerate = true) val idAction: Long = 0,
//    val type: String,
//    val date: LocalDateTime,
//    @ColumnInfo(name = "plant_id") val plantId: Long
//)
